xcdat/dataset.py

```diff
@@ -343,22 +348,14 @@ def decode_non_cf_time(dataset: xr.Dataset) -> xr.Dataset:
     ds = ds.assign_coords({time.name: decoded_time})
 
     if time_bounds is not None:
-        if units == "months":
-            data_bounds = [
-                [
-                    ref_date + rd.relativedelta(months=lower),
-                    ref_date + rd.relativedelta(months=upper),
-                ]
-                for [lower, upper] in time_bounds.data
-            ]
-        elif units == "years":
-            data_bounds = [
-                [
-                    ref_date + rd.relativedelta(years=lower),
-                    ref_date + rd.relativedelta(years=upper),
-                ]
-                for [lower, upper] in time_bounds.data
+        data_bounds = [
+            [
+                ref_dt_obj + rd.relativedelta(**{units: lower}),
+                ref_dt_obj + rd.relativedelta(**{units: upper}),
             ]
+            for [lower, upper] in time_bounds.data
+        ]
+
```

Comment:
This object for bounds should also be `cftime.datetime`. I am going to push a code to address this.